src/main/scala/zio/redis/Command.scala

```diff
@@ -0,0 +1,83 @@
+package zio.redis
+
+import java.time.Instant
+
+import zio.{ Chunk, IO }
+import zio.stream.Stream
+import zio.duration.Duration
+
+import scala.util.matching.Regex
+
+final case class Command[-In, +Out] private[redis] (
```

Comment:
How will pipelined commands be handled with this API? pipelined commands can either return nothing if they are queued in memory or they can be sent and will return QUEUED from Redis as a simple string, most Scala libraries I've seen do the ugly thing of returning null (as instances of the return type).

Reply:
To avoid returning meaningless responses, I believe we'll have to tackle pipelining separately. It might be long down the road, but I'll definitely open it up as an issue as soon as we have a working prototype.

Reply:
Another alternative (although I'm not suggesting that it's good in any way) is to return an `Id[Out] = Out` or an `Queued[Out] = Unit`, meaning that the client with implement the API with Id and the Pipeline client with Queued.
Is this ugly?

Honestly the best solution is to pipeline everything like the Haskell client, but I'm not sure this can be done using ZIO's Monad.

Reply:
If you're talking about hedis, I know what you mean. That might be nice option.

Reply:
```scala
trait Queued[-R, +E, +A] { 
  // map, zip, zipLeft, zipRight, etc.

  def run: ZIO[R, E, A]
}
```

Then you can form composite commands like:

```scala
(echo("foo") *> ping *> del("foo")).execute
```

which will be fully pipelined.

Hedis uses a terrible hack to achieve its magic "through" bind. We can't do that in Scala, but even if we could, I'd want to avoid that.